src/views/home/components/bridge-form/bridge-form.view.tsx

```diff
@@ -217,61 +137,88 @@ const BridgeForm: FC<BridgeFormProps> = ({
+    const areTokensPending = tokens?.some((tkn) => tkn.balance?.status === "pending");
+
+    if (selectedChains && tokens && areTokensPending) {
+      const getUpdatedTokens = (tokens: Token[] | undefined, updatedToken: Token) =>
+        tokens
+          ? tokens.map((tkn) => (tkn.address === updatedToken.address ? updatedToken : tkn))
+          : undefined;
+
+      setTokens((oldTokens) =>
+        !oldTokens
+          ? undefined
+          : oldTokens.map((tkn) => ({ ...tkn, balance: { status: "loading" } }))
+      );
+
+      tokens.map((token: Token) => {
+        getTokenBalance(token, selectedChains.from)
+          .then((balance): void => {
             callIfMounted(() => {
-              setBalanceTo(balance);
-            })
-          )
-          .catch((error) => {
+              const updatedToken: Token = {
+                ...token,
+                balance: {
+                  status: "successful",
+                  data: balance,
+                },
+              };
+
+              setTokens((currentTokens) => getUpdatedTokens(currentTokens, updatedToken));
+            });
+          })
+          .catch(() => {
             callIfMounted(() => {
-              notifyError(error);
-              setBalanceTo(undefined);
+              const updatedToken: Token = {
+                ...token,
+                balance: {
+                  status: "failed",
+                  error: "Couldn't retrieve token balance",
+                },
+              };
+
+              setTokens((currentTokens) => getUpdatedTokens(currentTokens, updatedToken));
             });
           });
-      } else {
-        getErc20TokenBalance({
-          chain: selectedChains.to,
-          tokenAddress: selectTokenAddress(token, selectedChains.to),
-          accountAddress: account,
-        })
-          .then((balance) =>
-            callIfMounted(() => {
-              setBalanceTo(balance);
-            })
-          )
-          .catch(() =>
-            callIfMounted(() => {
-              setBalanceTo(undefined);
-            })
-          );
-      }
+      });
+    }
+  }, [callIfMounted, defaultTokens, getTokenBalance, selectedChains, tokens]);
+
+  useEffect(() => {
+    // Load the balance of the selected token in the secondary network (To)
+    if (selectedChains && token) {
+      setBalanceTo({ status: "loading" });
+
+      getTokenBalance(token, selectedChains.to)
+        .then((balance) =>
+          callIfMounted(() => {
+            setBalanceTo({ status: "successful", data: balance });
+          })
+        )
+        .catch(() => {
+          callIfMounted(() => {
+            setBalanceTo({ status: "failed", error: "Couldn't retrieve token balance" });
+          });
+        });
     }
-  }, [selectedChains, account, token, getErc20TokenBalance, notifyError, callIfMounted]);
+  }, [callIfMounted, getTokenBalance, selectedChains, token]);
```

Comment:
We need to duplicate this effect for the From chain since we can no longer rely on the custom token to be part of `tokens` state.

```diff
diff --git a/src/views/home/components/bridge-form/bridge-form.view.tsx b/src/views/home/components/bridge-form/bridge-form.view.tsx
index a28f6b0..31519af 100644
--- a/src/views/home/components/bridge-form/bridge-form.view.tsx
+++ b/src/views/home/components/bridge-form/bridge-form.view.tsx
@@ -49,6 +49,9 @@ const BridgeForm: FC<BridgeFormProps> = ({
   const env = useEnvContext();
   const { getErc20TokenBalance, tokens: defaultTokens } = useTokensContext();
   const { connectedProvider } = useProvidersContext();
+  const [balanceFrom, setBalanceFrom] = useState<AsyncTask<BigNumber, string>>({
+    status: "pending",
+  });
   const [balanceTo, setBalanceTo] = useState<AsyncTask<BigNumber, string>>({ status: "pending" });
   const [inputError, setInputError] = useState<string>();
   const [selectedChains, setSelectedChains] = useState<SelectedChains>();
@@ -201,6 +204,25 @@ const BridgeForm: FC<BridgeFormProps> = ({
     }
   }, [callIfMounted, defaultTokens, getTokenBalance, selectedChains, tokens]);

+  useEffect(() => {
+    // Load the balance of the selected token in the primary network (From)
+    if (selectedChains && token) {
+      setBalanceFrom({ status: "loading" });
+
+      getTokenBalance(token, selectedChains.from)
+        .then((balance) =>
+          callIfMounted(() => {
+            setBalanceFrom({ status: "successful", data: balance });
+          })
+        )
+        .catch(() => {
+          callIfMounted(() => {
+            setBalanceFrom({ status: "failed", error: "Couldn't retrieve token balance" });
+          });
+        });
+    }
+  }, [callIfMounted, getTokenBalance, selectedChains, token]);
+
   useEffect(() => {
     // Load the balance of the selected token in the secondary network (To)
     if (selectedChains && token) {
@@ -252,8 +274,6 @@ const BridgeForm: FC<BridgeFormProps> = ({
     );
   }

-  const balanceFrom = tokens?.find((tkn) => tkn.address === token.address)?.balance;
-
   return (
     <form className={classes.form} onSubmit={onFormSubmit}>
       <Card className={classes.card}>
```

Reply:
We could also probably use the same effect.

Reply:
we can probably use the same effect

Reply:
Right! Done